src/main/java/com/melon/portfoliomanager/services/AssetService.java

```diff
@@ -71,5 +73,32 @@ public User validateUser(TransactionDto transactionDto) {
         return userList.get(0);
     }
 
+    public void sellStock(TransactionDto transactionDto) {
+
+        User user = validateUser(transactionDto);
+
+        List<PortfolioItem> portfolioItemList = portfolioItemRepository.findByUserIdAndCompanyName(user.getId(), transactionDto.getAssetSymbol());
+
+        PortfolioItem portfolioItem;
+
+        if (portfolioItemList.isEmpty()) {
+            throw new NotEnoughStocksToSell();
+        }
+
+        portfolioItem = portfolioItemList.get(0);
+
+        if (portfolioItem.getQuantity() < transactionDto.getQuantity()) {
+            throw new NotEnoughStocksToSell();
+        }
+
+        if (Objects.equals(portfolioItem.getQuantity(), transactionDto.getQuantity())) {
+            portfolioItemRepository.delete(portfolioItem);
+        } else {
+            portfolioItem.setQuantity(portfolioItem.getQuantity() - transactionDto.getQuantity());
```

Comment:
In the future, may be we should add additional column to the ```PortfolioItem``` database table for storing something like ```totalSoldPrice``` for easier calculation of some of the metrics for the analytics endpoint.
And then, add the sell price to the ```totalSoldPrice```.

Reply:
fixed